consensus/spos/bls/v2/subroundEndRound.go

```diff
@@ -544,7 +541,7 @@ func (sr *subroundEndRound) createAndBroadcastProof(signature []byte, bitmap []b
 }
 
 func (sr *subroundEndRound) createAndBroadcastInvalidSigners(invalidSigners []byte) {
-	if !sr.IsSelfLeader() {
+	if !sr.ShouldConsiderSelfKeyInConsensus() {
```

Comment:
TODO: we need to revisit this, as it could enable flooding

Reply:
done